packages/graphql/src/translate/cypher-builder/build-merge-statement.ts

```diff
@@ -159,13 +158,16 @@ export function buildMergeStatement({
     return joinStatements([mergeNodeStatement, onCreateSetQuery, ...onCreateStatements]);
 }
 
-function buildOnCreate(onCreate: Record<string, any>, varName: string): CypherStatement {
+function buildOnCreate(onCreate: Record<string, any>, varName: string, node?: Node | undefined): CypherStatement {
     const queries: string[] = [];
     const parameters = {};
 
     Object.entries(onCreate).forEach(([key, value]) => {
-        queries.push(`${varName}.${key} = $${generateParameterKey(`${varName}_on_create`, key)}`);
-        parameters[generateParameterKey(`${varName}_on_create`, key)] = value;
+        const nodeField = node?.primitiveFields.find((f) => f.fieldName === key);
+        const usedFieldName = nodeField ? nodeField.dbPropertyName || nodeField.fieldName : key;
+        queries.push(`${varName}.${usedFieldName} = $${generateParameterKey(`${varName}_on_create`, usedFieldName)}`);
+        const val = nodeField?.typeMeta.array ? [value] : value;
+        parameters[generateParameterKey(`${varName}_on_create`, usedFieldName)] = val;
```

Comment:
**nitpick**: There is a bit of duplicate code here,  consider using a variable for 
```ts
generateParameterKey(`${varName}_on_create`, usedFieldName)
```

so it can be also used in line 168